src/holder.ts

```diff
@@ -28,8 +36,13 @@ export class Holder {
    * @throws An error if the key binding JWT cannot be created.
    * @returns The key binding JWT.
    */
-  async getKeyBindingJWT(aud: string, nonce: string = generateNonce()): Promise<{ keyBindingJWT: JWT; nonce: string }> {
+  async getKeyBindingJWT(
+    aud: string,
+    nonceGenerator: NonceGenerator = nonceGeneratorCallbackFn(),
+  ): Promise<{ keyBindingJWT: JWT; nonce: string }> {
     try {
+      const nonce = nonceGenerator();
```

Comment:
@vijayshiyani my interpretations might be wrong, but I think the nonce is meant to be provided by the verifier
https://www.ietf.org/archive/id/draft-ietf-oauth-sd-jwt-vc-00.html#section-6.1-2.1.1

So the holder would have to include it as is in the kb jwt

Reply:
@ragnika correct, will fix it. thanks for identifying it.